caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/CadcMastResolver.java

```diff
@@ -97,12 +97,12 @@
 
     public CadcMastResolver() {
         try {
-            RegistryClient rc = new RegistryClient();
             Subject subject = AuthenticationUtil.getCurrentSubject();
```

Comment:
I guess I misssed this yesterday: the ctor should not do any init at all. Both the AuthMethod and the base URL lookup should happen in the toURL method and no values kept in member variables. That way the impl will adapt to changes in the data web service capabilities (even if the resolver is kept for long period of time and re-used).

I would make the same change in AdResolver - no member variables (just static) and do the service lookup in the toURL method.

Reply:
So we don't support public void setAuthMethod(AuthMethod authMethod) in AdResolver?